aiida_mlip/calculations/base.py

```diff
@@ -274,3 +255,77 @@ def prepare_for_submission(
         ]
 
         return calcinfo
+
+    def _define_architecture(self, cmd_line: dict) -> Union[dict, str]:
+        """
+        Find architecture in inputs or config file and add to command line if needed.
+
+        Parameters
+        ----------
+        cmd_line : dict
+            Dictionary containing the cmd line keys.
+
+        Returns
+        -------
+        cmd_line_updated: dict
+            Dictionary containing the cmd line keys updated with the architecture.
+        architecture: str
+            Architecture type, either from inputs/config or a default value ("mace_mp").
+        """
+        architecture = None
+        cmd_line_updated = cmd_line
```

Comment:
Are you wanting this to be a copy of `cmd_line`? 

At the moment, `cmd_line_updated is cmd_line == True` - they refer to the same object.

This means when you change `cmd_line_updated` after this, you change `cmd_line`, both in this function, and outside too (so you don't actually have to return the dictionary at all).

(This is also relevant in the other new function.)

Reply:
I was trying to make a copy yes (it should be cmd_line.copy() instead right?)
But I'm gonna change cmd_line anyway, so it might as well not be a copy... but I understood it is better to make copies from previous prs in which I was doing something similar?

Reply:
Yes, it should be `cmd_line.copy()` for a shallow copy.